### PR TITLE
UDP compiler plugin implementation

### PR DESCRIPTION
**Pull request by MohamedSabthar**
$title

### CODE REVIEW

File: udp-compiler-plugin-tests/src/test/resources/ballerina_sources/sample_package_10/bytes_service.bal [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import ballerina/udp as u;
+import sample_10.module as m;
+
+service on  new u:Listener(8000) {
```
Comment (Maninda) [inline udp-compiler-plugin-tests/src/test/resources/ballerina_sources/sample_package_10/bytes_service.bal:4]:
```suggestion
service on new u:Listener(8000) {
```

File: udp-compiler-plugin-tests/src/test/resources/ballerina_sources/sample_package_10/bytes_service.bal [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+   remote function onBytes(byte[] & readonly data) returns byte[] | u:Error? {
```
Comment (Maninda) [inline udp-compiler-plugin-tests/src/test/resources/ballerina_sources/sample_package_10/bytes_service.bal:6]:
```suggestion
   remote function onBytes(byte[] & readonly data) returns byte[]|u:Error? {
```

File: udp-compiler-plugin-tests/src/test/resources/ballerina_sources/sample_package_10/bytes_service.bal (hunk 1 above)
Comment (Maninda) [reply udp-compiler-plugin-tests/src/test/resources/ballerina_sources/sample_package_10/bytes_service.bal:4]: Change other locations as well.

File: udp-compiler-plugin-tests/src/test/resources/ballerina_sources/sample_package_10/bytes_service.bal (hunk 2 above)
Comment (Maninda) [reply udp-compiler-plugin-tests/src/test/resources/ballerina_sources/sample_package_10/bytes_service.bal:6]: Change other locations as well.

Comment (MohamedSabthar) [reply udp-compiler-plugin-tests/src/test/resources/ballerina_sources/sample_package_10/bytes_service.bal]: updated all resources.

File: udp-compiler-plugin-tests/src/test/resources/ballerina_sources/sample_package_10/bytes_service.bal (hunk 1 above)
Comment (MohamedSabthar) [reply udp-compiler-plugin-tests/src/test/resources/ballerina_sources/sample_package_10/bytes_service.bal]: updated all resources.

**Review by Maninda — APPROVED** [review]

File: udp-ballerina/build.gradle [hunk 3]
```diff
@@ -203,7 +206,7 @@ task ballerinaBuild {
             if (Os.isFamily(Os.FAMILY_WINDOWS)) {
                 commandLine 'cmd', '/c', "$balJavaDebugParam $distributionBinPath/bal.bat build -c ${testParams} ${debugParams} && exit %%ERRORLEVEL%%"
             } else {
-                commandLine 'sh', '-c', "$balJavaDebugParam $distributionBinPath/bal build -c ${testParams} ${debugParams}"
+                commandLine 'sh', '-c', "$balJavaDebugParam $distributionBinPath/bal build -c ${testParams} ${debugParams} && $distributionBinPath/bal push --repository=local"
```
Comment (aashikam) [inline udp-ballerina/build.gradle:209]: Was this change intentional?

Comment (MohamedSabthar) [reply udp-ballerina/build.gradle:209]: This is added to test the implementation in the local machine, ```&& $distributionBinPath/bal push --repository=local``` can be removed

Comment (aashikam) [reply udp-ballerina/build.gradle:209]: If it is added to test the implementation in the local machine, this shouldn't be here.

Comment (MohamedSabthar) [reply udp-ballerina/build.gradle:209]: +1, fixed in #96

Comment (Maninda) [reply udp-ballerina/build.gradle:209]: Merged it.
